File: DAIRemoteApp/app/src/main/java/com/example/dairemote_app/DisplayProfilesRecyclerAdapter.java
package com.example.dairemote_app;

import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class DisplayProfilesRecyclerAdapter extends RecyclerView.Adapter<DisplayProfilesRecyclerAdapter.OptionViewHolder> {
    private List<String> displayProfiles;

    public DisplayProfilesRecyclerAdapter(List<String> displayProfiles) {
        this.displayProfiles = displayProfiles;
    }

    @NonNull
    @Override
    public OptionViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.audio_item, parent, false);
        return new OptionViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull OptionViewHolder holder, int position) {
        if ("No display profiles".equals(displayProfiles.get(position))) {
            holder.button.setText("No Display Profiles");
            holder.button.setEnabled(false);
            holder.button.setBackgroundColor(Color.GRAY);
            holder.button.setOnClickListener(null); // Remove click listener
        } else {
            holder.button.setText(displayProfiles.get(position));
            holder.button.setBackgroundColor(Color.parseColor("#077063"));
            holder.button.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    MainActivity.connectionManager.SendHostMessage("DisplayConnect " + holder.button.getText().toString());
                }
            });
        }
    }

    @Override
    public int getItemCount() {
        return displayProfiles.size();
    }

    public void updateOptions(List<String> newOptions) {
        displayProfiles = newOptions;
        notifyDataSetChanged();
    }

    public static class OptionViewHolder extends RecyclerView.ViewHolder {
        Button button;

        public OptionViewHolder(@NonNull View itemView) {
            super(itemView);
            button = itemView.findViewById(R.id.audioDevicesButton);
        }
    }
}

File: DAIRemoteApp/app/src/main/java/com/example/dairemote_app/utils/DisplayProfilesRecyclerAdapter.kt
package com.example.dairemote_app.utils

import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.recyclerview.widget.RecyclerView
import com.example.dairemote_app.R

class DisplayProfilesRecyclerAdapter(
    private var displayProfiles: List<String>,
    private val connectionManager: ConnectionManager
) : RecyclerView.Adapter<DisplayProfilesRecyclerAdapter.OptionViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): OptionViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.interaction_list_item, parent, false)
        return OptionViewHolder(view)
    }

    override fun onBindViewHolder(holder: OptionViewHolder, position: Int) {
        if ("No display profiles" == displayProfiles[position]) {
            holder.button.text = "No Display Profiles"
            holder.button.isEnabled = false
            holder.button.setBackgroundColor(Color.GRAY)
            holder.button.setOnClickListener(null) // Remove click listener
        } else {
            holder.button.text = displayProfiles[position]
            holder.button.setBackgroundColor(Color.parseColor("#077063"))
            holder.button.setOnClickListener { connectionManager.sendHostMessage("DisplayConnect " + holder.button.text.toString()) }
        }
    }

    override fun getItemCount(): Int {
        return displayProfiles.size
    }

    fun updateOptions(newOptions: List<String>) {
        displayProfiles = newOptions
        notifyDataSetChanged()
    }

    class OptionViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var button: Button = itemView.findViewById(R.id.audioDevicesButton)
    }
}
